File: Backend/src/main/java/musiccube/entities/Role.java
package musiccube.entities;

import org.hibernate.annotations.NaturalId;

import javax.persistence.*;

@Entity
@Table(name = "roles")
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Enumerated(EnumType.STRING)
    @NaturalId
    @Column(name = "roleName")
    private RoleName name;

    // Constructors

    public Role() {
    }

    public Role(RoleName name) {
        this.name = name;
    }

    // Getters

    public int getId() {
        return id;
    }

    public RoleName getName() {
        return name;
    }

    // Setters

    public void setName(RoleName name) {
        this.name = name;
    }
}

File: src/app/Class/Role.ts
import { RoleName } from './RoleName';

export class Role {

    id: number;
    name: string;

    constructor(obj?: any) {
        this.id = (obj && obj.id ) || 0;
        this.name = (obj && obj.name) || null;
    }

    getId(): number {
        return this.id
    }

    getName(): string {
        return this.name;
    }

    setName(name: string) {
        this.name = name;
    }

}
